src/checks.py

```diff
@@ -79,16 +76,50 @@ def git_dir(self) -> str:
         return os.path.join(self.work_dir, "git", self.git_repo_name)
 
 
-def run_checks(state: State, checks: List[Callable[[State], Optional[str]]]) -> Report:
+class Check:
```

Comment:
Not an issue, I know it's already like this. But why is this a class and not a function?

Reply:
Because I wanted a way to store state on the check: the human-friendly check name and whether we should hide the check in the report if the check passed. In my mind function + state = class.

Or am I missing your point?

Totally open to refactoring this to be clearer.

Reply:
Yup, makes sense